src/intersections/lines.jl

```diff
@@ -43,14 +28,43 @@ end
 function intersectpoint(l1::Line, l2::Line)
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
+  λ₁, _ = intersectparameters(a, b, c, d)
+  a + λ₁ * (b - a)
+ end
+
+#= 
+Compute the parameters of the lines defined by the points a -> b and c -> d 
+so that the resulting points have minimal distance.
+
+The ranks help to identify the different types of intersection:
 
-  v1  = a - b
-  v2  = c - d
-  v12 = a - c
+Intersection: r == rₐ == 2
+Collinear: r == rₐ == 1
+No intersection: r != rₐ
+  No intersection and parallel:  r == 1, rₐ == 2
+  No intersection, skew lines: r == 2, rₐ == 3
+=#
+function intersectparameters(a::Point{N,T}, b::Point{N,T}, 
+                             c::Point{N,T}, d::Point{N,T}) where {N,T}
+  # solves the equation (approximately):
+  # a + λ₁ ⋅ v⃗₁ = c + λ₂ ⋅ v⃗₂,
+  # with v⃗₁ = b - a, v⃗₂ = d - c
+  A = [(b - a) (c - d)]
```

Comment:
notice that the comment and the expression differ, the second column has the sign flipped, can you please double check and make it consistent?

Reply:
everything is correct here, I thought this is clear from the context.
You have to bring the unknown lambdas on one side of the equation to solve them.
And you cannot redefine `v⃗₂` as it is the vector that points away from `c` with a length indicated by parameter `λ₂`
I'll add the comment line
``` code
# A = [v⃗₁ -v⃗₂]
```

Reply:
Yes, the more explicit the better. People reading this code for the first time will be more confident that things are the way they are for a reason and that the original author was super careful with minimal details.